actions.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"errors"
+	"github.com/Terag/kadok/security"
+	"github.com/bwmarrin/discordgo"
+	"strings"
+)
+
+// ExecuteAction is the function type to implement with an Action
+type ExecuteAction func(s *discordgo.Session, m *discordgo.MessageCreate) error
+
+// ActionResolver is the interface used by Kadok to resolve a call into an action
+type ActionResolver interface {
+	GetPermission() security.Permission
+	ResolveAction(call []string) (ActionResolver, ExecuteAction)
+}
+
+// Action structure containing information and functions regarding an action
+type Action struct {
+	// The permission required to use this action. Use the EmptyPermission if no permission is required
+	Permission security.Permission
+	// The sub actions in the hierarchy. They do not inherit the permission requirement
```

Comment:
would that means that there is no permission tree above the actions ?

As an example that means :

 * parent action _restricted to admin_
    * chlid action 1 _not restricted unless specified_
    * child action 2 _not restricted unless specified_ 

If my understanding is correct then it might lead to some issue if not documented well

Reply:
The example is correct.
An issue should be opened to address this point.

Reply:
Please see issue #23